### DISCUSSION TITLE
Encountering some sort of SSL error.

### DISCUSSION THREAD

**Discussion by shreyan01** [deepgram/community#937] [General help] [answered]
This is my code:
`async def transcribe_audio():
    try:
        headers = {
            "Authorization": f"Token {DEEPGRAM_API_KEY}",
            "Content-Type": "audio/mpeg",
        }
        with open("input_audio.mp3", "rb") as file:
            audio_data = file.read()
        params = {
            "diarize": "true",
            "model": "nova-2"
        }
        response = requests.post("https://api.deepgram.com/v1/listen", headers=headers, params=params, data=audio_data)
        response_json = response.json()
        with open("output.json", "w") as file:
            json.dump(response_json, file, indent=4)
        return JSONResponse(content={"message": "Transcription completed successfully"}, status_code=200)
    except Exception as e:
        logger.error("Error transcribing audio: %s", e)
        return JSONResponse(content={"message": f"Error transcribing audio: {e}"}, status_code=500)`
        
 And this is the error I'm encountering:
 `apps.backend.main:Error transcribing audio: HTTPSConnectionPool(host='api.deepgram.com', port=443): Max retries exceeded with url: /v1/listen?diarize=true&model=nova-2 (Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:2427)')))`

**Comment by jpvajda** [comment, marked as answer]
@shreyan01 

The error message you're seeing indicates an issue with the HTTPS connection between your application and Deepgram's API. Here’s a breakdown of what each part of the error likely means:

1. **`HTTPSConnectionPool(host='api.deepgram.com', port=443): Max retries exceeded`**:
   - This means that your application tried multiple times to connect to the Deepgram API at `api.deepgram.com` on the HTTPS port (443) but failed each time, eventually reaching the maximum number of allowed retries.

2. **`SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:2427)')`**:
   - The `SSLError` indicates that there's an SSL (Secure Sockets Layer) problem, specifically an `SSLEOFError`.
   - This error implies that the connection was closed unexpectedly during the SSL/TLS handshake, which is required to establish a secure connection. "EOF" (End of File) here indicates that the connection was prematurely terminated, violating the expected SSL protocol.

3. **Potential Causes**:
   - **Network issues**: There might be network disruptions or firewall rules preventing your app from reaching the Deepgram API.
   - **SSL/TLS incompatibility**: This could happen if the client (your app) and the server (Deepgram API) don’t support the same version of SSL/TLS.
   - **Expired or invalid certificate**: If your client doesn’t recognize Deepgram’s SSL certificate, it might terminate the connection.
   - **Rate-limiting or temporary server issue**: Deepgram could be temporarily unavailable or rate-limiting connections, though this typically returns a different error.

4. **Steps to Fix**:
   - **Update libraries**: Make sure your HTTP client (e.g., `requests` in Python) and SSL libraries (like `OpenSSL`) are up to date.
   - **Check network configuration**: Ensure no firewall or proxy is blocking the connection.
   - **Verify Deepgram’s API status**: Check [Deepgram’s status page](https://status.deepgram.com/#) 
   - ** Rate Limiting**  Check out our [documentation for rate limits.](https://developers.deepgram.com/reference/api-rate-limits)
   - **Set retries and timeouts**: Use a retry mechanism in your code with exponential backoff if network instability is suspected.
